DatadogTrace/Sources/OpenTelemetry/OTelSpanBuilder.swift

```diff
@@ -0,0 +1,140 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import OpenTelemetryApi
+
+internal class OTelSpanBuilder: OpenTelemetryApi.SpanBuilder {
+    var tracer: DatadogTracer
+    var spanName: String
+    var spanKind = SpanKind.client
+    var attributes: [String: OpenTelemetryApi.AttributeValue]
+    var startTime: Date?
+    var active: Bool
+    var parent: Parent
```

Comment:
blocking/ It's all a mutable state modified by user and it lacks synchronisation. `@ReadWriteLock` might do the job.

Reply:
I would argue that this needs to thread safe given the purpose of the builder. When a customer is building the span using the `SpanBuilder` it must be all done from the same thread. Once they start the span, it can be move around different threads.

Reply:
On the same hypothesis https://opentelemetry.io/docs/specs/otel/trace/api/#concurrency